portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,34 @@ function addFunFacts() {
     const factsContainer = document.getElementById('facts-container');
     factsContainer.innerText = fact;
 }
+
+
+/** Fetches the current date from the server and adds it to the page. */
+// async function showServerTime() {
+//   const responseFromServer = await fetch('/date');
+//   const textFromResponse = await responseFromServer.text();
+
+//   const dateContainer = document.getElementById('date-container');
+//   dateContainer.innerText = textFromResponse;
+// }
+
+/** Fetches stats from the server and adds them to the page. */
+async function getFacts() {
+  const responseFromServer = await fetch('/facts');
+  // The json() function returns an object that contains fields that we can
+  // reference to create HTML.
+  const stats = await responseFromServer.json();
+
+  // Pick a random fact.
+  const quotes = stats[Math.floor(Math.random() * stats.length)];
+
+  const statsListElement = document.getElementById('quote-container');
+  statsListElement.innerHTML = quotes;
```

Comment:
I missed this earlier, but I think this should probably be using `.innerText` instead of `.innerHTML`. When setting `.innerText`, any HTML characters (e.g. `<`) will be escaped properly such that they don't interfere with rendering the page. However, when using `.innerHTML`, if the data coming from the server contains characters which alter the flow of the HTML, they will potentially prevent your page from rendering properly.

Reply:
Thank you. That could be the reason why my escape characters are not working as I intended.

Reply:
And yes that fixed it.